File: projects/scala/src/main/scala/ws/autocomplete/results/Disambiguation.scala
package ws.autocomplete.results

case class Disambiguation(uriA: String,
                          labelA: String,
                          bs: List[PageElement],
                          available: Boolean)
  extends Result {
  override def prettyPrint(): String = {
    val t = new StringBuilder
    t ++= s"d: $labelA($uriA)"
    bs.map {
      case PageElement(uri, label, in, a) =>
        t ++= s"\n   $label ($uri: $in, available: $a)"
    }
    t.toString()
  }
  override def searchLabel(): String = labelA
  override def pageUri(): String = uriA
  override def displayLabel(): String = labelA
  override def isAvailable(): Boolean = available
}

File: projects/website/app/search/bar/completion/result/disambiguation.ts
import {Entity} from "./entity";
import {Result} from "./result";

export class Disambiguation extends Result{
    constructor(public label: String,
                public uri: String,
                public entities: Array<Entity>) {
    }

    public isDisambiguation(): boolean {
        return true;
    }

    public asDisambiguation(): Disambiguation {
        return this;
    }
}
